vnext/Scripts/Tfs/Make-AnyCPU-RefAssemblies.ps1

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+param(
+	[string] $BuildRoot = "$SourceRoot\vnext\target",
+	[string] $TargetRoot = "$SourceRoot\vnext\target"
+)
+
+
+# Fix ref Assemblies to be AnyCpu
+
+# The reason for doing this here is that csc maintains the architecture of the ref assembly to match the implementation. It doesn't 
+# automatically tag the ref as 'anycpu'. We use the ref assembly as the 'ref' in our nuget package. When the csc compiler
+# compiles for other specific platforms like Arm, the compiler will fail if a ref assembly is marked as x64.
+# The fix is to ensure that the 'ref' in the nuget package is marked as 'AnyCPU'.
+
+# There unfortunately is no flag to csc, and no single tool to do this. CorFlags.exe does not have support to change this...
+# This is a lot more straight-forward than trying to create an AnyCpu version of this assembly, update the solution, tests and add
+# another platform to the CI matrix, not to mention the steps needed to ensure AnyCpu ran for the x64 build to test the nuget package
+# in PR validation....
+
+# Therefore this step will simply ildasm and ilasm the reference assembly and store it as which strips the bitness and generates anycpu msil.
+# These reference assemblies are just regular assemblies but with empty bodies
+
+ForEach ($refFolder in (Get-ChildItem -Path  $BuildRoot -Recurse -Include "ref"))
+{ 
+	ForEach ($refAsm in Get-ChildItem -Path $refFolder -Recurse  -Include "*.dll" ) 
+	{
+		$outputFolder=([System.IO.Path]::GetDirectoryName($refAsm)).Replace($BuildRoot, $TargetRoot) + ".anycpu";
+		$outputAsm = [System.IO.Path]::Combine($outputFolder, [System.IO.Path]::GetFileName($refAsm));
+		$outputIl = [System.IO.Path]::ChangeExtension($outputAsm, ".il");
+
+		mkdir $outputFolder -ErrorAction SilentlyContinue
+		& "${Env:ProgramFiles(x86)}\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\ildasm.exe" $refAsm /ALL "/OUT=$outputIl"
```

Comment:
nit: This is a potential break for future .net sdk versions and windows sdk versions i.e. v11 vs netfx5 etc...
Fine for now, but maybe we should have an explicit existance check for this path with a very specific error message about that this should be made dynamic the first time it is hit.

Reply:
Good feedback to the person who wrote this 😉. https://github.com/microsoft/react-native-windows/pull/6529

For this PR I pulled this logic out of `Layout-Headers`, so that we can make the ref assemblies before uploading artifacts, then not need to deal with it on the machine combining artifacts into the NuGet package. This is to get away from the previous setup where we designated one build machine to run Layout-Headers, and would be blocked until it was complete.